v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/GCSToSourceDb.java

```diff
@@ -247,6 +248,52 @@ public interface Options extends PipelineOptions, StreamingOptions {
     String getRunIdentifier();
 
     void setRunIdentifier(String value);
+
+    @TemplateParameter.GcsReadFile(
+        order = 15,
+        optional = true,
+        description = "Custom jar location in Cloud Storage",
+        helpText =
+            "Custom jar location in Cloud Storage that contains the custom transformation logic for processing records"
+                + " in reverse replication.")
+    @Default.String("")
+    String getTransformationJarPath();
+
+    void setTransformationJarPath(String value);
+
+    @TemplateParameter.Text(
+        order = 16,
+        optional = true,
+        description = "Custom class name",
+        helpText =
+            "Fully qualified class name having the custom transformation logic.  It is a"
+                + " mandatory field in case transformationJarPath is specified")
+    @Default.String("")
+    String getTransformationClassName();
+
+    void setTransformationClassName(String value);
+
+    @TemplateParameter.Text(
+        order = 17,
+        optional = true,
+        description = "Custom parameters for transformation",
+        helpText =
+            "String containing any custom parameters to be passed to the custom transformation class.")
+    @Default.String("")
+    String getTransformationCustomParameters();
+
+    void setTransformationCustomParameters(String value);
+
+    @TemplateParameter.Boolean(
+        order = 18,
+        optional = true,
+        description = "Write filtered events to GCS",
+        helpText =
+            "This is a flag which if set to true will write filtered events from custom transformation to GCS.")
```

Comment:
what path do the filtered events get written to ?

Reply:
files are written to the same GCS bucket/filtered events where the processed records by the reader job are written to. Had a discussion around this with @aksharauke and she mentioned that it should help the users compare the filtered records as they have similar name as the processed record file

Reply:
I will add a info log around the same for the users to be informed